### ISSUE TITLE
[C+ payment required, Situ out until May] [HOLD for payment 2024-04-03] [$500] Desktop - Login - Unable to enter the 2FA code or exit the opened screen on desktop

### ISSUE THREAD

**Issue by lanitochka17** [Expensify/App#34177]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:**  1.4.23-0
**Reproducible in staging?:** Y
**Reproducible in production?:** Y
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Expensify/Expensify Issue URL:**
**Issue reported by:** Applause - Internal Team
**Slack conversation:**

## Action Performed:
*Prerequisites: User must be logged into web app as User A.*
1. Launch the Desktop app
2. Enter an email for an existing account with 2FA (User b)
3. Go to the email inbox and locate the magic code
4. Copy the Magic link and modify it to staging
5. On the web, open a new tab and navigate to the staging magic link
6. Confirm that the user is prompted to open the desktop app
7. Click on the "Open app" option

## Expected Result:
When the desktop is launched by the magic code link, it glitches, briefly shows a skeleton and then returns to the first login screen

## Actual Result:
When the desktop is launched by the magic code link, there's just a blocker screen asking to enter the magic code, with no field and no option to exit the screen

## Workaround:
Unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android: Native
- [ ] Android: mWeb Chrome
- [ ] iOS: Native
- [ ] iOS: mWeb Safari
- [ ] MacOS: Chrome / Safari
- [x] MacOS: Desktop

## Screenshots/Videos

Add any screenshot/video evidence

</details> 

https://github.com/Expensify/App/assets/78819774/f07d7998-7fff-41b6-9f79-6dc2ef507cbd

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~0183573e4ff04521ad</li>
        <li>Upwork Job ID: 1744794019492372480</li>
        <li>Last Price Increase: 2024-03-06</li>
<li>Automatic offers: </li>
<ul>
<li>situchan | Contributor | 28115507</li>
</ul></li>
    </ul>
</details>

<details><summary>Issue Owner</summary>Current Issue Owner: @trjExpensify</details>

**Comment by ikevin127** [comment]
# Proposal

## Please re-state the problem that we are trying to solve in this issue

When the desktop is launched by the magic code link, there's just a blocker screen asking to enter the magic code, with no field and no option to exit the screen.

## What is the root cause of that problem?

This happens because in `ValidateLoginPage/index.website.js` there's no logic that specifically handles 2FA login flow from the desktop app where there's no multi-tab functionality like on web.

For example the flow on web is as follows:
- we input the email for login
- the UI changes showing us the magic code input
- we click the magic code link from the email
- a new tab opens rendering the same 2FA screen telling the user to input their 2FA code in the tab where the magic code input was showing previously before opening the magic link

## What changes do you think we should make in order to solve the problem?

My solution is to create a desktop only library `@libs/desktopTwoFactorRedirect/index.desktop.js` (with index.js `noop`) where we check if `(autoAuthState === CONST.AUTO_AUTH_STATE.JUST_SIGNED_IN && !isSignedIn)` is `true` and if so we call ` Navigation.navigate();` then **return;**

Notes: 
- `autoAuthState === CONST.AUTO_AUTH_STATE.JUST_SIGNED_IN` confirms that we passed the magic code step of the login process which is one of the conditions that if `true` we're shown the 2FA screen
- `!isSignedIn` confirms that we're not signed-in yet as there's one last step and that's the 2FA validation
- calling `Navigation.navigate();` when above conditions are true will navigate the user to 2FA code validation screen

This lib function takes in `autoAuthState` and `isSignedIn` as arguments and will be called at the top of the first `useEffect` inside the `ValidateLoginPage/index.website.js` component.

**Comment by ikevin127** [comment]
## Updated [proposal](https://github.com/Expensify/App/issues/34177#issuecomment-1884116162)
- updated solution `if` to account for the edge case where `autoAuthState === CONST.AUTO_AUTH_STATE.NOT_STARTED` (after logout) which would still cause the 2FA screen lock

**Comment by situchan** [comment]
I can take over

**Comment by peterdbarkerUK** [comment]
Note on the above, @situchan is added as the *reviewer*

**Comment by situchan** [comment]
> 6. Confirm that the user is prompted to open the desktop app

I stopped here. I am not prompted to open desktop app

<img width="1016" alt="2fa" src="https://github.com/Expensify/App/assets/108292595/645a1d4a-7730-4f55-a09d-798ae3a0efb7">

**Comment by situchan** [comment]
Not overdue

**Comment by ikevin127** [comment]
> I stopped here. I am not prompted to open desktop app

This is the 'blocker screen' you should see on the desktop app once transitioned to it from web - if following the steps and opening the magic link (adding staging to it) and clicking the 'Open New Expensify' once the prompt shows, while using a 2FA account.

I can still reproduce this on latest staging (v1.4.33-3).

<details>
<summary>MacOS: Desktop</summary>

https://github.com/Expensify/App/assets/56457735/c6068798-6694-4387-a03a-a39666ed63ae

</details>

**Comment by trjExpensify** [comment]
I'm getting the blocking screen on web: 

<img width="1511" alt="image" src="https://github.com/Expensify/App/assets/16232057/1f7ee8f6-bccc-4b5c-8dbf-8acef5257070">

Steps: 

1. Sign-out of both desktop and web 
2. Sign-in on web, prompted for magic code 
3. Go to email
4. Copy the URL and modify to "staging."
5. Blocking 2FA page appears.

**Comment by trjExpensify** [comment]
Now.. if I actually click the link (like most would instead of copying it), I get a different result. 

1. Sign-out of both desktop and web
2. Sign-in on web, prompted for magic code
3. Go to email
4. Click the link
5. Land on this page and prompted to open desktop

<img width="1511" alt="image" src="https://github.com/Expensify/App/assets/16232057/967e6cae-e156-41d2-a708-789111d2a1d7">

6. Click to open desktop and land here: 
<img width="1511" alt="image" src="https://github.com/Expensify/App/assets/16232057/62316263-0f66-4acc-8e20-7db84e4391be">

7. Click "just sign in here" and land here to enter my 2FA code:
<img width="1511" alt="image" src="https://github.com/Expensify/App/assets/16232057/91df045e-6f3e-401e-98b2-0401aa1795d5">

8. Enter my 2FA code and I'm signed in.

**Comment by ikevin127** [comment]
Thanks for looking into this! ✌

This looks like a pretty rare edge case on Desktop. I can only reproduce this with the production desktop app installed and changing the magic link to staging.

As for using the production magic link while having the staging desktop app installed - seems to work as expected as per your latest comment.

The blocking screen on web is expected IMO as when somebody with 2FA attempts login on web and clicks the magic link, this opens a new window where we see the 2FA blocking screen - this assumes you still have your initial tab open where you attempted login - where now you have to introduce the 2FA code since that's what the blocking screen in the other tab is telling you.

Finally, even though this Desktop issue can be hard to reproduce - if we want to block the 2FA flow for Desktop app then [my colleague's solution](https://github.com/Expensify/App/issues/34177#issuecomment-1884332533) would do.

Otherwise, if we want to allow the 2FA flow on Desktop then [my proposal](https://github.com/Expensify/App/issues/34177#issuecomment-1884116162) still stands. It will allow the 2FA flow for users on the desktop app via the web prompt - users will be able to go through with it as on Desktop there's no new tab functionality like on web - so it has to happen within the desktop app like demonstrated in the solution's result video:

<details>
<summary>MacOS: Desktop</summary>

https://github.com/Expensify/App/assets/56457735/7d429304-1d87-4ac2-895f-292ab662f7ba

</details>

**Comment by trjExpensify** [comment]
Thanks, @ikevin127. If you use the production desktop app and a production magic link, does everything work okay?

**Comment by trjExpensify** [comment]
Ah okay, cool. @NikkiWines @johnmlee101 you folks were close to this implementation, perhaps you can help us with what's expected here? 

Summary of steps to repro: 

1. User has 2FA enabled 
2. User has the desktop app installed (signed out). 
3. User goes to the desktop app to sign-in, gets sent a magic link on email 
4. Clicks the link in the email and gets taken to NewDot on web 
5. Clicks the modal prompt to switch to desktop
6. Gets hit with the 2FA full page blocking form

**Comment by trjExpensify** [comment]
They get this one, judging by the video above in [this comment](https://github.com/Expensify/App/issues/34177#issuecomment-1926989637): 

<img width="815" alt="image" src="https://github.com/Expensify/App/assets/16232057/83e1132f-63c5-4790-aef5-e5ff7497bc68">

**Comment by johnmlee101** [comment]
Hmm, and the account has 2FA enabled right?

**Comment by johnmlee101** [comment]
But yeah, I think its a bug. It should take them to the 2FA input screen if they have it enabled

**Comment by trjExpensify** [comment]
For the avoidance of doubt, don't suppose you could type out the steps like [here](https://github.com/Expensify/App/issues/34177#issuecomment-1927025557), so we can follow exactly what's expected?

**Comment by johnmlee101** [comment]
1. User has 2FA enabled
2. User has the desktop app installed (signed out).
3. User goes to the desktop app to sign-in, gets sent a magic link on email
4. Clicks the link in the email and gets taken to NewDot on web
5. Clicks the modal prompt to switch to desktop
**6. Desktop is presented with the 2FA Code input screen**
**7. User enters 2FA code**
**8. User is now taken into the authenticated product on Desktop.**

**Comment by ikevin127** [comment]
@johnmlee101 I would add a note:
- make sure that the environment matches web / desktop app

Meaning if 2FA account attempts login from production desktop app then clicking on the email magic link directly is ok as the link will start with https://new.expensifi.com...

On staging (web / desktop app), the `staging.` should be appended to the email magic link before navigating to it and getting the prompt (as mentioned in the OP) ✌️

**Comment by NikkiWines** [comment]
All the magic code links direct to prod, I believe - regardless of whether the request is initiated on staging web, staging desktop, prod web, prod desktop, etc.

**Comment by ikevin127** [comment]
Correct!

If I'm not mistaking, when we get the magic link email it's always the production domain link.

This can lead to confusion when trying to reproduce this issue as can be seen in the comments above 😅

**Comment by NikkiWines** [comment]
> > Thanks, @ikevin127. If you use the production desktop app and a production magic link, does everything work okay?

> @trjExpensify Unfortunately no 😞
> I just tried what you suggested on latest production (version 1.4.35-7) and it reproduces exactly as the OP suggests.

Given this behavior, it seems to me it's for sure a bug. We should do one of the following:

- Ensure we don't block the user from entering their 2FA code (via one of the proposals from above, if still applicable)
- Don't prompt the user to open the desktop app when using a magic link. Looking back on when we [first implemented this](https://github.com/Expensify/App/pull/14443/files#diff-4a06a628c4f2fa783e590af0b888f09723c061691252843b9e5270df3ae95d45R26-R28) we specifically avoided linking to the desktop app in this scenario. 

As a bonus, we should probably fix the links on the backend so that they point to the appropriate environment and don't need to be manually modified to work on staging. That's out of scope for this issue though.

**Comment by johnmlee101** [comment]
> Don't prompt the user to open the desktop app when using a magic link. Looking back on when we [first implemented this](https://github.com/Expensify/App/pull/14443/files#diff-4a06a628c4f2fa783e590af0b888f09723c061691252843b9e5270df3ae95d45R26-R28) we specifically avoided linking to the desktop app in this scenario.

Ohhhhh thanks for the context forgot all about that. Yeah Desktop support in this way was never explicitly made when we were developing for this for a reason

**Comment by trjExpensify** [comment]
If we do one of these:

>We should do one of the following:
>
> - Ensure we don't block the user from entering their 2FA code (via one of the proposals from above, if still applicable)
> - Don't prompt the user to open the desktop app when using a magic link. Looking back on when we [first implemented this](https://github.com/Expensify/App/pull/14443/files#diff-4a06a628c4f2fa783e590af0b888f09723c061691252843b9e5270df3ae95d45R26-R28) we specifically avoided linking to the desktop app in this scenario.

... won't they still run into this:

1. User has 2FA enabled
2. User has the desktop app installed (signed out).
3. User goes to the **desktop app to sign-in**, gets sent a magic link on email
4. Clicks the link in the email and gets taken to **NewDot on web**
5. Does not get prompted to go to the desktop app, but nevertheless, hits this screen minus the prompt and can't authenticate: 

<img width="973" alt="image" src="https://github.com/Expensify/App/assets/16232057/9d89bcab-0a0f-4bc4-aa20-ddf1600b6816">

**Comment by NikkiWines** [comment]
> Does not get prompted to go to the desktop app, but nevertheless, hits this screen minus the prompt and can't authenticate:

Hmm, I tested this using prod just now and wasn't blocked on authenticating. They should have the field to manually enter the magic code in the desktop app at this point.

**Comment by NikkiWines** [comment]
I think ideally they'd just see the the following, unless I'm misunderstand the flow: 
- On web, they'd get the `Here is your magic code` view with the copy notifying them to enter the code where they originally requested it from. 
- On desktop, they'd see the view for entering the magic code.

Like so: 

https://github.com/Expensify/App/assets/16747822/bdfb9191-81d9-49bb-b129-d13f4ed355bf

**Comment by trjExpensify** [comment]
Cool yeah, same page! So I think we're taking a pretty big leap of faith that a person who lands on the magic code form on web will instinctively know to switch over to the Desktop app. It's a pretty confusing message given that it is the same device they requested on, right? 

So the prompt to switch to Desktop when installed is actually quite helpful to reorientate to them in that regard. To the point where I'm wondering if it's a better experience to [make this work](https://github.com/Expensify/App/issues/34177#issuecomment-1927611244).

**Comment by NikkiWines** [comment]
Hhm yeah, the copy could be better for sure. I personally don't think it's too disruptive for the user but I'm also totally down with us fixing the flow so that the deep link to desktop works 👍

**Comment by trjExpensify** [comment]
>but I'm also totally down with us fixing the flow so that the deep link to desktop works 👍

Okay cool, @ikevin127 do you want to update to reflect that and then @situchan can review?

**Comment by ikevin127** [comment]
@trjExpensify Sure, I can handle this. I'd say the RCA and solution of my proposal are the same, will fix the issue just as discussed.

Is there something I'm missing - that you suggest adding to my current proposal ?

**Comment by trjExpensify** [comment]
Ah cool, I'll defer to @situchan to take a look as is then! Perhaps @johnmlee101 or @NikkiWines would like to be the secondary reviewer from our team internally when ready. 

>I don't have the full context of this issue but i request to put it on hold in favour of https://github.com/Expensify/App/pull/35359 if this both are related, We migrated the files to typescript and the PR will be merged in coming days

Seems like this might be wise though. What does everyone think?

**Comment by ikevin127** [comment]
Not sure, my solution requires a small addition within `src/pages/ValidateLoginPage/index.website.js` which I don't see as part of the migration within the mentioned PR 👀

Anybody else ?

cc @GandalfGwaihir

**Comment by situchan** [comment]
Need to hold for TS migration

**Comment by trjExpensify** [comment]
Cool, @situchan can you jump on the proposal review then, please? Thanks!

**Comment by trjExpensify** [comment]
Bump, @situchan. What's your ETA on reviewing?

**Comment by situchan** [comment]
reviewing today/tomorrow based on latest discussion

**Comment by situchan** [comment]
still evaluating proposals. eta: end of week

**Comment by trjExpensify** [comment]
@situchan what's going on here? It has been 2 weeks now. 

P.s @greg-schroeder I'm moving this into wave6 as LOW. It's a sign-in bug that can have an impact on anyone with 2FA, but given submitters is the largest population of users, wave6 makes sense to me.

**Comment by greg-schroeder** [comment]
I guess so, but it's not really in the scope of any of our upcoming releases, so I'll leave it at `LOW` until a home becomes clear or it just gets resolved

**Comment by situchan** [comment]
@ikevin127 @tienifr can you please share test branches?
It should fix both 2fa and no 2fa cases.

**Comment by situchan** [comment]
And please update your proposals with the correct root cause.
Especially, why is not user prompted to open the desktop app? (both 2fa and no 2fa)

**Comment by ikevin127** [comment]
The review took so long to get started that navigation refactor was performed in the meantime and my initial proposal doesn't work anymore, specifically the `Navigation.navigate()` part which doesn't work anymore if no arguments are passed as per this comment here https://github.com/Expensify/App/issues/35782#issuecomment-1927837173:

> Also Navigation.navigate() without any parameters doesn't seem to work after navigation refactor.

And those parts of the old logic in the ValidateLoginPage were updated by the following PR https://github.com/Expensify/App/pull/35850 to account for the recent navigation refactor since `Navigation.navigate()` without any parameters doesn't seem to work anymore.

I'm looking into the logic and if I can find a way to solve the issue just as my initial proposal did, before navigation refactor, then I'll get back and update my proposal. If I don't come back with any update then it means I'm out of the race for this issue 🏳

**Comment by situchan** [comment]
awaiting proposals based on latest codebase

**Comment by trjExpensify** [comment]
Any luck, @ikevin127?

**Comment by situchan** [comment]
still awaiting proposals

**Comment by WojtekBoman** [comment]
Hey! @ikevin127 Could you share the branch with your changes? I'll try to reproduce it and check what's wrong

**Comment by ikevin127** [comment]
@WojtekBoman Sorry I don't have a branch, all I have is my [initial proposal](https://github.com/Expensify/App/issues/34177#issuecomment-1884116162) - which worked as described in solution (see videos) before the new navigation.

Also this is another [issue](https://github.com/Expensify/App/issues/35782) and its PR https://github.com/Expensify/App/pull/35850 that changed the old implementation to use `Navigation.goBack()` because after the new navigation `Navigation.navigate()` did not work anymore like it used to be.

So I guess what would help here is to better understand how we can call a similar navigation function o achieve the same as we did previously using `Navigation.navigate()` - because if how the new bottom tab navigator is structured currently, calling navigate with home route while non-authenticated does not seem to work at all anymore.

Another outcome that would help here is to have a way of opening web prompt with magic link code in desktop app, process the code in the desktop app and:
- if the account has 2FA, redirect to 2FA code page
- if account doesnt have 2FA, perform login

**Comment by trjExpensify** [comment]
@WojtekBoman any thoughts on the above?

**Comment by adamgrzybowski** [comment]
Hi! I will try to help

**Comment by trjExpensify** [comment]
Going to assign @adamgrzybowski so he can help @ikevin127 here.

**Comment by adamgrzybowski** [comment]
@ikevin127 Not sure if I understand the problem completely but you can try to use `goBack()` to pop the screen and then `navigate()`

**Comment by ikevin127** [comment]
## Updated [proposal](https://github.com/Expensify/App/issues/34177#issuecomment-1884116162)
- changed initial solution code of stack popping method, since new navigation logic was implemented and the old code did not work anymore

@situchan As per [#34177 (comment)](https://github.com/Expensify/App/issues/34177#issuecomment-1953601218) -> 

>And please update your proposals with the correct root cause.

Please explain what you mean by `correct root cause` so I know what other details need to be added to the proposal, if any.
Below are detailed steps + video to consistently reproduce and test the issue / proposed solution:

1. Open incognito browser tab.
2. Navigate to https://staging.new.expensify.com.
4. Login with any account (once logged in, a prompt to Open in New Expensify / Electron [for dev] should appear if you have the desktop app installed / running in dev).
5. Once logged in, open the Desktop app (dev or staging - depending on what you test).
6. Login with 2FA account.
7. Copy the magic link received via email as a result of login within the Desktop app.
8. Open a new incognito tab next to the already logged in account.
9. Paste the magic link and add `staging.` as a prefix, eg. email magic link: `https://new.expensify.com/v/15879736/903350` will become `https://staging.new.expensify.com/v/15879736/903350`.
10. Press enter and navigate to the staging magic link, this will open the prompt to Open in New Expensify / Electron (for dev).

<details>
<summary>Video</summary>

https://github.com/Expensify/App/assets/56457735/28f256e6-acf1-4168-a32d-0a0e7c693bc5

</details>

> [!note]
> 1. In order to repeat the cycle, make sure to clear all onyx keys from Application > IndexedDB, otherwise you will stay locked on the 2FA lock screen.
> 2. The video above was filmed at the time of this comment -> proving the issue is still reproducible.
> 3. [Here](https://github.com/ikevin127/Expensify/tree/fix/34177)'s the test branch w/ my proposal's solution applied.

@adamgrzybowski Thanks for the input, after doing some more digging I was able to figure out how to re-write the fix which worked before the new navigation implementation.

Since just `navigate()` or `goBack()` + `navigate()` wouldn't work anymore because of the new navigation implementation, I figured that what `navigate()` alone was doing before was popping the stack, so I did the same now but more directly by using `navigationRef.current?.dispatch(StackActions.popToTop())`.

**Comment by trjExpensify** [comment]
Thanks, @ikevin127. Perhaps @adamgrzybowski & @situchan can check out that [test branch](https://github.com/ikevin127/Expensify/tree/fix/34177)?

**Comment by adamgrzybowski** [comment]
@ikevin127 Thanks for the great description and instructions for reproduction 🙇 ! Makes my life much easier 😄 

I tested it and it works 🎉 I think it should work fine in this case. 

Do you have a PR open so I can look into diff easier? 

BTW it looks like the solution you found `navigationRef.current?.dispatch(StackActions.popToTop())` may be handy in some other use cases. I talked with @WojtekBoman and we think that it may be a good idea to create a function for that in [Navigation.ts](https://github.com/Expensify/App/blob/3d4504e54154540629477cfc6ae0e897a09e8a5d/src/libs/Navigation/Navigation.ts) just to make reusing it easier.

**Comment by ikevin127** [comment]
@adamgrzybowski Thanks for the great feedback! I'm glad the instructions made testing this easy 🙌

> Do you have a PR open so I can look into diff easier?

I don't have a PR because @situchan did not review the proposals yet and I am not assigned to this as a Contributor, hence opening a PR is not recommended. But I can do it if this is an exception - let me know!

I created the branch in order to make testing easier for the reviewers.

> I talked with **WojtekBoman** and we think that it may be a good idea to create a function for that in [Navigation.ts](https://github.com/Expensify/App/blob/3d4504e54154540629477cfc6ae0e897a09e8a5d/src/libs/Navigation/Navigation.ts) just to make reusing it easier.

If assigned as a Contributor to this issue I'm fully on board w/ adding this in the PR as a function within `Navigation` and using it both in my proposal and other parts of the app (where needed) 🚀

What do you think about `Navigation.popToTop()` as a name ? Open to more suggestive names!

**Comment by situchan** [comment]
@adamgrzybowski you can check diff here:
https://github.com/Expensify/App/compare/main...ikevin127:Expensify:fix/34177

**Comment by mountiny** [comment]
@situchan @adamgrzybowski seems like you are happy to go with this @ikevin127 proposal, any concerns?

**Comment by situchan** [comment]
sure, let's assign @ikevin127

**Comment by adamgrzybowski** [comment]
@mountiny  it's good ✅

**Comment by mountiny** [comment]
Thanks

**Comment by ikevin127** [comment]
PR https://github.com/Expensify/App/pull/38142 is ready for review! 🚀

**Comment by ikevin127** [comment]
> We had discussion with @WojtekBoman. The function that he needs will require to do popToTop on two navigators (root and bottom tab).

Just solved a conflict on the PR and I ended up using newly added `Navigation.resetToHome()` as it's more robust and it accomplishes the same thing for our issue (double-checked).

**Comment by adamgrzybowski** [comment]
Great to hear that it this function works for your use case as well! Thanks for testing this 🙇

**Comment by ikevin127** [comment]
@situchan 🙋‍♂️ It's been 3 days since the PR is ready for review - is there anything blocking the process ? Please let us know!

**Comment by situchan** [comment]
I reviewed and commented yesterday. Going ahead today.

**Comment by ikevin127** [comment]
👋 From what I know the `Weekly` label on an issue means that the proposals should be reviewed on a weekly basis.

Does that also apply to the PR reviewing process ?

And if the answer is yes - given the following turn of events:
- reviewer attempts review for the first time
- there are some conflicts, reviewer tags author to let them know
- conflicts solved the same day by author

Next time when the PR is checked again could be up to one week later ?

I'm trying to figure out what do the time related labels mean in the context of the PR being ready for review and what are the expected timelines for completion.

cc @stitesExpensify @mountiny

**Comment by mountiny** [comment]
I dont think the priority label influences how the PR should be reviewed or proposals should be reviewed. That should still be ideally ever day or once in 48 hours at worst.

Made it daily to avoid confusion

**Comment by mountiny** [comment]
$500 to @ikevin127 and $500 to @situchan

**Comment by ikevin127** [comment]
> [!note]
> I did receive an Upwork Contributor offer for this issue on 12th of March despite [this message](https://github.com/Expensify/App/issues/34177#issuecomment-1992691904), but I cannot accept it as it errors (see error below) - therefore I declined it and will await BZ manual offer 🙌
<img width="693" alt="Screenshot 2024-04-02 at 17 38 25" src="https://github.com/Expensify/App/assets/56457735/d3a84363-d0e4-41b9-a08d-67c4ffce3866">

**Comment by trjExpensify** [comment]
Sounds good! Payment summary as follows: 

- $500 to @ikevin127 for the fix 
- $500 to @situchan for the C+ review

Offers sent.

**Comment by trjExpensify** [comment]
Paid you!

**Comment by trjExpensify** [comment]
Situ is out until May, so this is going to sit until then. Updated the title to make that clear.

**Comment by trjExpensify** [comment]
Still awaiting Situ's return.

**Comment by trjExpensify** [comment]
Still waiting on Situ so I can pay this.

**Comment by situchan** [comment]
Sorry for late. I think job expired. Can you please send new offer?

**Comment by trjExpensify** [comment]
Offer sent!

**Comment by trjExpensify** [comment]
Situ paid, closing!
